src/resources/2019/DiabolicPizza.ts

```diff
@@ -0,0 +1,92 @@
+import { autosellPrice, getWorkshed, toInt, visitUrl } from "kolmafia";
+import { get as getModifier } from "../../modifier";
+import { have as haveItem } from "../../lib";
+import { $item, $monster } from "../../template-string";
+import { clamp } from "../../utils";
+
+/**
+ * @returns true if the Diabolic pizza cube is installed in workshed.
+ */
+export function installed(): boolean {
+  return getWorkshed() === $item`diabolic pizza cube`;
+}
+
+/**
+ * @returns true if the Diabolic pizza cube is installed in workshed or in inventory.
+ */
+export function have(): boolean {
+  return installed() || haveItem($item`diabolic pizza cube`);
+}
+
+/**
+ * @param items One or more items to be checked.
+ * @returns true if all items are a valid pizza ingredients.
+ */
+export function validIngredients(...items: Item[]): boolean {
```

Comment:
doesn't this also need to check npcPrice?

Reply:
Oh yes that's right.